.pylintrc

```diff
@@ -133,7 +133,9 @@ disable=print-statement,
         too-few-public-methods,
         too-many-arguments,
         missing-docstring,
-        protected-access
+        protected-access,
+        line-too-long,
+        no-self-use
```

Comment:
Is this suppression needed? I tried taking it out locally and didn't get any pylint errors under Python 2.7.9 or 3.6.4.

Reply:
When I run pylint using that file, I get the "protected_access" and "no-self-use" for my unit tests.

The protected access I have to violate to fully test some functions, the no-self-use is a result of the "Base<blahblah>Test" class I use to have a single location where I define the DxlClient config behavior.

"line-too-long" must have been copied from a separate project, I don't believe I have any need for it.

Reply:
Ah, ok. I do get a `no-self-use` error if I modify the `LintCommand.run()` method in the setup.py file to also lint the test files:

> self.announce("Running pylint for library source files and tests", level=distutils.log.INFO)
subprocess.check_call(["pylint", "dxlepoclient", **"tests"**] + glob.glob("*.py"))

Might be good to add that into this PR as well?

In this case, I believe the warning is due to the `BaseEpoClientTest.create_client()` method not making any references to self variables. Rather than suppressing the warning in the pylint file, I think we could fix this by just declaring `create_client` as a `@staticmethod`. What do you think?

For `line-too-long`, yeah, it would be nice to not have to suppress it. I've only left that in for projects where there were a ton of source code changes that I'd otherwise have had to make in order to shorten existing lines of code. Generally speaking, though, I think it is nice for code style to have pylint nag us about using shorter lines.

Reply:
- Removed extraneous line-too-long and regex pattern.
- create_client on base class is now static

I'll put these changes in the other unit test PRs too, and commit them